src/messages.rs

```diff
@@ -202,19 +202,26 @@ pub struct GroupList {
 pub struct SignedMessage {
     /// A request or response type message.
     content: RoutingMessage,
-    /// The lists of the groups involved in routing this message, in chronological order.
-    grp_lists: Vec<GroupList>,
+    /// Nodes sending the message (those expected to sign it)
+    sending_nodes: BTreeSet<PublicId>,
```

Comment:
Maybe; I didn't want to think that far ahead. Can we leave it until we get there?